### PR TITLE
new(falco): add buffer_format_base64 option, deprecate -b

### PR DESCRIPTION
**Pull request by LucaGuerra**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area engine

**What this PR does / why we need it**:

Add `buffer_format_base64` option which behaves like `-b` `--print-base64`, works in the config file or with `-o`.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
new(falco): add buffer_format_base64 option, deprecate -b
```

### CODE REVIEW

**Comment by LucaGuerra** [comment]
@FedeDP done!

**Review by FedeDP — DISMISSED** [review]
/approve

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by sgaist — DISMISSED** [review]

**Review by FedeDP — APPROVED** [review]
/approve

**Comment by FedeDP** [comment]
Rebase & fix conflict? Then we will merge it :)

**Review by FedeDP — APPROVED** [review]
/approve

File: userspace/falco/app/actions/init_inspectors.cpp [hunk 1]
```diff
@@ -26,7 +26,18 @@ using namespace falco::app;
...
 
 static void init_syscall_inspector(falco::app::state& s, std::shared_ptr<sinsp> inspector) {
-	inspector->set_buffer_format(s.options.event_buffer_format);
+	sinsp_evt::param_fmt event_buffer_format = sinsp_evt::PF_NORMAL;
+	if(s.options.print_base64) {
+		falco_logger::log(falco_logger::level::WARNING,
+		                  "The -b/--print-base64 option is deprecated and will be removed. Use -o "
+		                  "buffer_format_base64=true instead.");
+		event_buffer_format = sinsp_evt::PF_BASE64;
+	}
```
Comment (sgaist) [inline userspace/falco/app/actions/init_inspectors.cpp:35]: Question for a followup: shouldn't that warning rather be printed when parsing the options ?

That way, you can keep all the deprecation related code in one single place.

**Review by sgaist — APPROVED** [review]

**Pull request merged**